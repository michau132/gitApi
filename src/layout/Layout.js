import React, { Fragment } from 'react';
import styled, { createGlobalStyle } from 'styled-components';

const StyledWrapper = styled.div`
  display: grid;
  height: 100vh;
`;


const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Lora:400,700&subset=latin-ext');
  html {
    box-sizing: border-box;
  }
  *, *:before, *:after {
    box-sizing: inherit;
    margin: 0;
  }
  body {
    padding: 0;
    margin: 0;
    font-size: 16px;
    line-height: 1.7;
    font-family: 'Lora';
    background-color: #f8f8f8;
  }
  a {
    text-decoration: none;
    color: inherit;
  }
`;

const Layout = ({ children }) => (
  <Fragment>
    <GlobalStyle />
    <StyledWrapper>
      {children}
    </StyledWrapper>
  </Fragment>
);

export default Layout;
