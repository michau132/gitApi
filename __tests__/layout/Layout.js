import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Layout from '../../src/layout/Layout';

describe('testing Layout component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Layout />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('to match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
