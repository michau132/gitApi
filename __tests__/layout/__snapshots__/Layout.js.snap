// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Layout component to match snapshot 1`] = `
<Fragment>
  <GlobalStyleComponent
    suppressMultiMountWarning={false}
  />
  <styled.div />
</Fragment>
`;
